pkg/controller/utils/component.go

```diff
@@ -407,6 +410,51 @@ func ensureOSSchedulingRestrictions(obj client.Object, osType rmeta.OSType) {
 	modifyPodSpec(obj, f)
 }
 
+// setStandardSelectorAndLabels will set the k8s-app and app.kubernetes.io/name Labels on the podTemplates
+// for Deployments and Daemonsets. If there is no Selector specified a selector will also be added
+// that selects the k8s-app label.
+func setStandardSelectorAndLabels(obj client.Object) {
+	var podTemplate *v1.PodTemplateSpec
+	var name string
+	switch obj.(type) {
+	case *apps.Deployment:
+		d := obj.(*apps.Deployment)
+		name = d.ObjectMeta.Name
+		if d.ObjectMeta.Labels == nil {
+			d.ObjectMeta.Labels = make(map[string]string)
+		}
+		d.ObjectMeta.Labels["k8s-app"] = name
```

Comment:
Guess we need to keep this around for now, but wondering if we should eventually migrate off it to the app.kubernetes.io label and remove this altogether.

Reply:
We certainly need to keep it until we switch over the matchSelector and I'm not sure if we can just switch that over on upgrade. I don't see much harm in keeping it.